src/event/event.rs

```diff
@@ -174,6 +174,17 @@ impl Event {
         sys::event::is_lio(&self.inner)
     }
 
+    /// Returns the OS specific event.
+    ///
+    /// The returned type depends on the OS and is thus not guaranteed. Portable
+    /// programs should not depends on this function and should only use the
+    /// addtional information store in the OS specific event as an optimisation.
+    #[cfg(all(unix, feature = "os-poll", feature = "os-ext"))]
+    #[cfg_attr(docsrs, doc(cfg(all(unix, feature = "os-poll", feature = "os-ext"))))]
+    pub fn as_raw_event(&self) -> &sys::Event {
```

Comment:
This binds the mio public API to a libc version. If the goal is to provide access to platform specific data, we can define accessors on the current `Event` struct.